gapis/api/templates/vulkan_gfx_api_extras.tmpl

```diff
@@ -209,158 +209,123 @@ bool Vulkan::replayCreateVkDeviceImpl(Stack* stack, size_val physicalDevice,
+      return true;
     }
+    stack->push<size_val>(static_cast<size_val>(instance));
+    stack->push<VkAllocationCallbacks const*>(pAllocator);
+    return callVkDestroyInstance(~0, stack, false);
   }
 ¶
   bool Vulkan::replayUnregisterVkInstance(Stack* stack) {
     auto instance = static_cast<VkInstance>(stack->pop<size_val>());
-    if (stack->isValid()) {
-      GAPID_DEBUG("replayUnregisterVkInstance(%zu)", instance);
-      mVkInstanceFunctionStubs.erase(instance);
-      auto& pdevMap = mIndirectMaps.VkPhysicalDevicesToVkInstances;
-      for (auto it = pdevMap.begin(); it != pdevMap.end();) {
-        if (it->second == instance) {
-          it = pdevMap.erase(it);
-        } else {
-          ++it;
-        }
+    GAPID_DEBUG("replayUnregisterVkInstance(%zu)", instance);
+    mVkInstanceFunctionStubs.erase(instance);
+    auto& pdevMap = mIndirectMaps.VkPhysicalDevicesToVkInstances;
+    for (auto it = pdevMap.begin(); it != pdevMap.end();) {
+      if (it->second == instance) {
+        it = pdevMap.erase(it);
+      } else {
+        ++it;
       }
-      return true;
-    } else {
-      GAPID_WARNING("Error during calling function replayUnregisterVkInstance");
-      return false;
     }
+    return true;
   }
 ¶
   bool Vulkan::replayRegisterVkDevice(Stack* stack) {
     auto createInfo = stack->pop<VkDeviceCreateInfo*>();
     auto device = static_cast<VkDevice>(stack->pop<size_val>());
     auto physical_device = static_cast<VkPhysicalDevice>(stack->pop<size_val>());
-    if (stack->isValid()) {
-      GAPID_DEBUG("replayRegisterVkDevice(%zu, %zu, %p)", physical_device, device, createInfo);
-      if (mVkDeviceFunctionStubs.count(device) > 0) {
-        // TODO(antiagainst): The same device handle returned from the driver. Is this possible?
-        return false;
-      }
-      mIndirectMaps.VkDevicesToVkPhysicalDevices[device] = physical_device;
-      auto instance = mIndirectMaps.VkPhysicalDevicesToVkInstances[physical_device];
-      auto& stubs = mVkDeviceFunctionStubs[device];
-      {{range $c := AllCommands $}}
-        {{if (Macro "IsIndirected" "Command" $c "IndirectOn" "VkDevice")}}
-          {{$name := Macro "CmdName" $c}}
-          stubs.{{$name}} = LazyResolved<{{Template "C++.FunctionPtrType" $c}}>(
-            [instance, device]() -> void* { return core::GetVulkanDeviceProcAddress(instance, device, "{{$name}}"); });
-        {{end}}
-      {{end}}
-      // Get all queues for this device and bind them.
-      for (auto queue : getVkQueues(stubs.vkGetDeviceQueue, device, createInfo)) {
-        mIndirectMaps.VkQueuesToVkDevices[queue] = device;
-      }
-      return true;
-    } else {
-      GAPID_WARNING("Error during calling function replayRegisterVkDevice");
+    GAPID_DEBUG("replayRegisterVkDevice(%zu, %zu, %p)", physical_device, device, createInfo);
+    if (mVkDeviceFunctionStubs.count(device) > 0) {
+      // TODO(antiagainst): The same device handle returned from the driver. Is this possible?
```

Comment:
This comment can be removed, this is not actually possible.

Reply:
Can the if block be removed as well as the comment then? This isn't my code.

Reply:
Yes, it is illegal for a handle to be duplicated.